pilot/pkg/serviceregistry/kube/controller/ambient/workloads.go

```diff
@@ -134,12 +134,13 @@ func (a *index) workloadEntryWorkloadBuilder(
 		policies := a.buildWorkloadPolicies(ctx, authorizationPolicies, peerAuths, meshCfg, wle.Labels, wle.Namespace)
 		var waypoint *Waypoint
 		if wle.Labels[constants.ManagedGatewayLabel] != constants.ManagedGatewayMeshControllerLabel {
-			waypoint = fetchWaypointForWorkload(ctx, waypoints, namespaces, wle.ObjectMeta)
+			// TODO: report waypoint status
```

Comment:
in queue.go in the reconcile function for StatusQueue, I see reporter.patcher and other references to reporter. Maybe that's the reporting we're referencing in this todo?

```go
type StatusQueue struct {
	queue controllers.Queue

	// reporters is a mapping of unique controller name -> status information.
	// Note: this is user facing in the fieldManager!
	reporters map[string]statusReporter
}

// statusItem represents the objects stored on the queue
type statusItem struct {
	Key      string
	Reporter string
}
```

would this todo then really just require some initialization of object of type statusReporter to be added to the reporters map eventually; which would consist of the controller name and status info at this point in time? or is there more (or less) to it?

Reply:
Yeah we would need to do a `statusqueue.Register` for `Workloads` and the associated logic around `getConditions` I think